cmd/vshn-postgres-func/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/urfave/cli/v2"
+	"github.com/vshn/appcat-comp-functions/pkg"
+	vp "github.com/vshn/appcat-comp-functions/pkg/functions/vshn-postgres-func"
+	vshnpostgres "github.com/vshn/appcat-comp-functions/pkg/functions/vshn-postgres-func"
+)
+
+func main() {
+	app := newApp()
+	err := app.Run(os.Args)
+	// If required flags aren't set, it will return with error before we could set up logging
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func newApp() *cli.App {
+	app := &cli.App{
+		Name:    vp.AI.AppName,
+		Usage:   vp.AI.AppLongName,
+		Version: fmt.Sprintf("%s, revision=%s, date=%s", vp.AI.Version, vp.AI.Commit, vp.AI.Date),
+		Action:  runAll,
+		After:   printFuncIO,
+		Before:  setup,
+		Flags: []cli.Flag{
+			pkg.NewLogLevelFlag(),
+			pkg.NewLogFormatFlag(),
+		},
+		Commands: []*cli.Command{},
+	}
+	return app
+}
+
+func runAll(ctx *cli.Context) error {
+
+	for _, cmd := range ctx.App.Commands {
+		if cmd.Name != "help" {
+			err := cmd.Run(ctx)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func printFuncIO(ctx *cli.Context) error {
+	log := logr.FromContextOrDiscard(ctx.Context)
+
+	funcIO, err := pkg.GetFunctionIO(ctx.Context)
+	if err != nil {
+		return err
+	}
+
+	return pkg.PrintFunctionIO(funcIO, log)
+}
+
+func setup(ctx *cli.Context) error {
+
+	err := pkg.SetupLogging(vp.AI, ctx)
+	if err != nil {
+		return err
+	}
+
+	_ = pkg.LogMetadata(ctx, vshnpostgres.AI)
+
+	funcIO, err := pkg.GetFunctionIO(ctx.Context)
+	if err != nil {
+		return err
+	}
+
+	ctx.Context = context.WithValue(ctx.Context, pkg.KeyFuncIO, funcIO)
```

Comment:
I like this idea, this way we make the whole boilerplate around the CLI framework much smaller. Looking into this.

Reply:
@glrf I've implemented your idea, what do you think? The boilerplate code is now reduced quite a bit.

I guess we could get rid of even more boilerplate. But that's probably something for the future.

Reply:
That looks better yes. I'm not a fan of passing `cli.Context` out of main though. At first I was very confused since when `context.Context` has a method `String`...

But otherwise looks OK to me